ext/standard/basic_functions.c

```diff
@@ -1722,16 +1727,13 @@ PHPAPI void php_free_shutdown_functions(void) /* {{{ */
 PHP_FUNCTION(register_shutdown_function)
 {
 	php_shutdown_function_entry entry;
-	zval *params = NULL;
-	uint32_t param_count = 0;
 	bool status;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "f*", &entry.fci, &entry.fci_cache, &params, &param_count) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "f*", &entry.fci, &entry.fci_cache, &entry.fci.params, &entry.fci.param_count) == FAILURE) {
```

Comment:
`entry.fci.params` will point to the VM call stack, and needs to be duplicated as it outlives the register_shutdown_function() call

Reply:
Ahhh that's indeed an issue. Okay here it makes sense to realloc. :)